app/models/operatingsystems/redhat.rb

```diff
@@ -25,9 +25,22 @@ def pxe_type
     "kickstart"
   end
 
+  def pxe_file_names(medium_provider)
+    if medium_provider.try(:architecture_name) =~ /^[Ss]390/
```

Comment:
In Ruby 2.7 comparing with nil can raise a warning I think. How about
```suggestion
    if medium_provider&.architecture_name&.match?(/^[Ss]390/)
```